workflow/rules/bwa-mem.smk

```diff
@@ -0,0 +1,48 @@
+# vim: syntax=python tabstop=4 expandtab
+# coding: utf-8
+
+__author__ = "Jonas Almlöf"
+__copyright__ = "Copyright 2021, Jonas Almlöf"
+__email__ = "jonas.almlof@scilifelab.uu.se"
+__license__ = "GPL-3"
+
+
+rule bwa_mem:
+    input:
+        fastq=["Pre-alignment/merge/{sample}_R1.fq.gz", "Pre-alignment/merge/{sample}_R2.fq.gz"],
+    output:
+        bam="Alignment/bwa-mem/{sample}.bam",
+    log:
+        log="Logs/Aligment/bwa-mem/{sample}.log",
+    params:
+        index=config["references"]["genome"],
+        extra=config["bwa-mem"]["extra"],
+        sort=config["bwa-mem"].get("sort", "samtools")
+        sort_order=config["bwa-mem"].get("sort_order", "coordinate"),
+        sort_extra="-@ " + str(config["bwa-mem"]["threads"]),
+    threads: config["bwa-mem"]["threads"]
+    #benchmark:
```

Comment:
I am personally not a fan of commented code. But you probably use it at a later stage, right?

Reply:
Do we want to include benchmarking in each tool and make a statistik for it? It might be nice :)

Reply:
I think we should try to add benchmark settings, if one don't want to use it one don't have to activate it during your analysis.

Reply:
maybe set it as 
`benchmark:
    benchmark/alignment/bwa-mem/{sample}.txt
`

Reply:
Are these `tsv`-files? Maybe we could be explicit about that 😉

Reply:
It is a .tsv file!
There could be other tsv-files generated by the rules. I suggest that we add .benchmark to the name to make i clearer:
benchmark: "alignment/bwa_mem/{sample}_{unit}.benchmark.tsv"

Reply:
good point!